### PR TITLE
Enable high concurrency mode

### PR DESCRIPTION
**Pull request by hwupathum**
### Related issue
https://github.com/wso2/product-is/issues/21513

### Summary
This pull request introduces a new feature to enable high concurrency mode and updates the related configuration files accordingly. The most important changes include adding a new parameter for concurrency type, modifying the performance script to handle this new parameter, and updating the CloudFormation templates to support high concurrency mode.

### Enhancements to Performance Script:
* Added `concurrency_type` and `enable_high_concurrency` variables in `common/start-performance.sh` to support high concurrency mode. [[1]](diffhunk://#diff-62b3c0a71a56c74a0854a4fe0bfa1f0798389a9005cef227fa3262945f70271cR29) [[2]](diffhunk://#diff-62b3c0a71a56c74a0854a4fe0bfa1f0798389a9005cef227fa3262945f70271cR43)
* Updated the `getopts` string and case statement to include the new `-r` option for concurrency type. [[1]](diffhunk://#diff-62b3c0a71a56c74a0854a4fe0bfa1f0798389a9005cef227fa3262945f70271cL102-R104) [[2]](diffhunk://#diff-62b3c0a71a56c74a0854a4fe0bfa1f0798389a9005cef227fa3262945f70271cR127-R129)
* Modified the `run_performance_tests_options` to include `concurrency_type` and added logic to enable high concurrency mode based on the concurrency type. [[1]](diffhunk://#diff-62b3c0a71a56c74a0854a4fe0bfa1f0798389a9005cef227fa3262945f70271cL192-R197) [[2]](diffhunk://#diff-62b3c0a71a56c74a0854a4fe0bfa1f0798389a9005cef227fa3262945f70271cR247-R251)
* Updated the `create_stack_command` to pass the `EnableHighConcurrencyMode` parameter to the CloudFormation stack.

### Updates to CloudFormation Templates:
* Added `EnableHighConcurrencyMode` parameter and removed `DBAllocationStorage` and `DBIops` parameters in `four-node-cluster/4-node-cluster.yml`. [[1]](diffhunk://#diff-89848d365f523d2c1dadfa3377acb240b1578cf856ec737696c553eb28ee5958R30) [[2]](diffhunk://#diff-89848d365f523d2c1dadfa3377acb240b1578cf856ec737696c553eb28ee5958L40-L41) [[3]](diffhunk://#diff-89848d365f523d2c1dadfa3377acb240b1578cf856ec737696c553eb28ee5958L50-L51) [[4]](diffhunk://#diff-89848d365f523d2c1dadfa3377acb240b1578cf856ec737696c553eb28ee5958R703-R708) [[5]](diffhunk://#diff-89848d365f523d2c1dadfa3377acb240b1578cf856ec737696c553eb28ee5958L734-L741) [[6]](diffhunk://#diff-89848d365f523d2c1dadfa3377acb240b1578cf856ec737696c553eb28ee5958L771-L778)
* Updated the mappings and resources in `four-node-cluster/4-node-cluster.yml` to use the new high concurrency mode settings. [[1]](diffhunk://#diff-89848d365f523d2c1dadfa3377acb240b1578cf856ec737696c553eb28ee5958L277-R274) [[2]](diffhunk://#diff-89848d365f523d2c1dadfa3377acb240b1578cf856ec737696c553eb28ee5958L288-R285) [[3]](diffhunk://#diff-89848d365f523d2c1dadfa3377acb240b1578cf856ec737696c553eb28ee5958L299-R296) [[4]](diffhunk://#diff-89848d365f523d2c1dadfa3377acb240b1578cf856ec737696c553eb28ee5958L309-R306) [[5]](diffhunk://#diff-89848d365f523d2c1dadfa3377acb240b1578cf856ec737696c553eb28ee5958R821-R831)

### Similar Updates to Other Templates:
* Applied the same changes to `single-node/single-node.yaml` and `three-node-cluster/3-node-cluster.yml` to support high concurrency mode. [[1]](diffhunk://#diff-5e8c2e11dcceddf85224ae6ba7952e1894e6f4ece11803775e3756f3fd021c80R27) [[2]](diffhunk://#diff-5e8c2e11dcceddf85224ae6ba7952e1894e6f4ece11803775e3756f3fd021c80L37-L38) [[3]](diffhunk://#diff-5e8c2e11dcceddf85224ae6ba7952e1894e6f4ece11803775e3756f3fd021c80L47-L48) [[4]](diffhunk://#diff-5e8c2e11dcceddf85224ae6ba7952e1894e6f4ece11803775e3756f3fd021c80L222-R219) [[5]](diffhunk://#diff-5e8c2e11dcceddf85224ae6ba7952e1894e6f4ece11803775e3756f3fd021c80L233-R230) [[6]](diffhunk://#diff-5e8c2e11dcceddf85224ae6ba7952e1894e6f4ece11803775e3756f3fd021c80L244-R241) [[7]](diffhunk://#diff-5e8c2e11dcceddf85224ae6ba7952e1894e6f4ece11803775e3756f3fd021c80L254-R251) [[8]](diffhunk://#diff-5e8c2e11dcceddf85224ae6ba7952e1894e6f4ece11803775e3756f3fd021c80R515-R520) [[9]](diffhunk://#diff-5e8c2e11dcceddf85224ae6ba7952e1894e6f4ece11803775e3756f3fd021c80L546-L553) [[10]](diffhunk://#diff-5e8c2e11dcceddf85224ae6ba7952e1894e6f4ece11803775e3756f3fd021c80L583-L590) [[11]](diffhunk://#diff-5e8c2e11dcceddf85224ae6ba7952e1894e6f4ece11803775e3756f3fd021c80R633-R643) [[12]](diffhunk://#diff-e4d7ddc3c354002f96893fcf19de5b8a5b1e7ea172af6dbcc2400f7c4ac65bdfR30) [[13]](diffhunk://#diff-e4d7ddc3c354002f96893fcf19de5b8a5b1e7ea172af6dbcc2400f7c4ac65bdfL40-L41) [[14]](diffhunk://#diff-e4d7ddc3c354002f96893fcf19de5b8a5b1e7ea172af6dbcc2400f7c4ac65bdfL50-L51) [[15]](diffhunk://#diff-e4d7ddc3c354002f96893fcf19de5b8a5b1e7ea172af6dbcc2400f7c4ac65bdfL251-R248) [[16]](diffhunk://#diff-e4d7ddc3c354002f96893fcf19de5b8a5b1e7ea172af6dbcc2400f7c4ac65bdfL262-R259) [[17]](diffhunk://#diff-e4d7ddc3c354002f96893fcf19de5b8a5b1e7ea172af6dbcc2400f7c4ac65bdfL273-R270)

These changes collectively enable the high concurrency mode feature, providing better performance for high-load scenarios.

### CODE REVIEW

File: common/start-performance.sh [hunk 1]
```diff
@@ -239,6 +244,11 @@ else
     exit 1
 fi
 
+# Enable high concurrency mode if the concurrency type contains 1000 or more
+if [[ $concurrency_type =~ ^([0-9]{4}-[0-9]{3}|[0-9]{3}-[0-9]{4}|[0-9]{4}-[0-9]{4})$ ]]; then
```
Comment (Sachin-Mamoru) [inline common/start-performance.sh:248]: what would be the value for concurrency_type?

Comment (hwupathum) [reply common/start-performance.sh:248]: Its the value assigned from jenkins from. Its in the format xxx-xxxx

Comment (hwupathum) [reply common/start-performance.sh]: Renamed vairable with https://github.com/wso2/performance-is/pull/201/commits/66337b94469e2592335f0fb4006584ae7e581ba8

**Review by Sachin-Mamoru — APPROVED** [review]

**Comment by hwupathum** [comment]
The IOPS for higher concurrency mode was decided by running the perf tests on 3000 concurrency and identifying the highest IOPS recorded in AWS RDS

**Pull request merged**